vmware_aria_operations_integration_sdk/mp_build.py

```diff
@@ -108,48 +124,138 @@ def build_subdirectories(directory: str) -> None:
         shutil.move(os.path.join(directory, file), dir_path)
 
 
-def get_registry_components(docker_registry: str) -> Tuple[str, str]:
-    components = docker_registry.split("/")
-    host = components[0]
-    path = "/".join(components[1:])
-    return host, path
+def is_valid_registry(container_registry: str, **kwargs: Any) -> bool:
+    try:
+        if _is_docker_hub_registry_format(container_registry):
+            if "registry_username" not in kwargs:
+                kwargs["registry_username"] = prompt("Enter Docker Hub username: ")
+
+            if "registry_password" not in kwargs:
+                kwargs["registry_password"] = prompt("Password: ", is_password=True)
+            login(**kwargs)
+
+            container_registry = (
+                container_registry
+                if container_registry.startswith("docker.io")
+                else f"docker.io/{container_registry}"
+            )
 
+        login(container_registry=container_registry, **kwargs)
 
-def is_valid_registry(docker_registry: str, **kwargs: Any) -> bool:
-    try:
-        login(docker_registry, **kwargs)
     except LoginError:
         return False
 
     return True
 
 
+def _is_docker_hub_registry_format(registry: Optional[str]) -> bool:
+
+    if not registry:
+        return False
+    # should match namespace/repo or docker.io/namespace/repo
+    # namespace must be between 4 and 30 characters long, and can only contain numbers and lowercase letters"
+    # repos must contain at least two characters, can't start or end with _ . -, can't contain uppercase letters
+    pattern = r"^(?:docker\.io\/[a-z0-9]{4,30}|[a-z0-9]{4,30})\/[a-z0-9]+[a-z0-9._-]*[a-z0-9]+$"
+
+    return bool(re.match(pattern, registry))
+
+
+def _tag_and_push(
+    image: Image,
+    container_registry_arg: Optional[str],
+    config_file: str,
+    manifest: dict,
+    adapter_kind_key: str,
+    docker_client: DockerClient,
+    **kwargs: Any,
+) -> tuple[str, str, str]:
+
+    logger.error(f"container registry arg: {container_registry_arg}")
+    container_registry = container_registry_arg
+    if not container_registry:
+        container_registry = get_config_value(
+            CONFIG_CONTAINER_REGISTRY_KEY, config_file=config_file
+        )
+    if not container_registry:
+        container_registry = get_config_value(
+            CONFIG_FALLBACK_CONTAINER_REGISTRY_KEY, config_file=config_file
+        )
+
+    # we want to keep track of the original value, so we can update it if nesessary
+    original_value = container_registry
+
+    digest = ""
+    should_prompt = container_registry is None
```

Comment:
IMO, we should prompt anytime the push fails because of an invalid tag, _except_ if the tag was specified via the command line argument.